File: src/codeforces/set2/set21/set214/set2141/c/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var n int
	fmt.Scan(&n)
	res := solve(n)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	fmt.Fprintln(writer, len(res))
	for _, cur := range res {
		fmt.Fprintln(writer, cur)
	}
}

func solve(n int) []string {

	pushBack := func(i int) string {
		return fmt.Sprintf("pushback a[%d]", i-1)
	}

	var res []string
	d := 1
	for l := 1; l <= n; l++ {
		if d == 1 {
			r := l
			for r <= n {
				res = append(res, pushBack(r))
				res = append(res, "min")
				r++
			}
			// r > n
			d = -1
		} else {
			r := n
			// d = -1
			// r = n
			for r > l {
				res = append(res, "min")
				res = append(res, "popback")
				r--
			}
			// l = r 但是还没有查询
			// res = append(res, "min")
			res = append(res, "min")
			d = 1
		}

		if l < n {
			res = append(res, "popfront")
		}
	}

	return res
}

File: src/codeforces/set2/set21/set214/set2141/c/solution.kt
import java.io.BufferedInputStream
import java.lang.StringBuilder

private class FastScanner {
    private val input = BufferedInputStream(System.`in`)
    private val buffer = ByteArray(1 shl 16)
    private var len = 0
    private var ptr = 0

    private fun readByte(): Int {
        if (ptr >= len) {
            len = input.read(buffer)
            ptr = 0
            if (len <= 0) return -1
        }
        return buffer[ptr++].toInt()
    }

    fun nextInt(): Int {
        var c = readByte()
        while (c <= 32) c = readByte()
        var sign = 1
        if (c == '-'.code) {
            sign = -1
            c = readByte()
        }
        var res = 0
        while (c > 32) {
            res = res * 10 + (c - '0'.code)
            c = readByte()
        }
        return res * sign
    }
}

private fun solve(n: Int): List<String> {
    val res = ArrayList<String>()
    var d = 1
    for (l in 1..n) {
        if (d == 1) {
            var r = l
            while (r <= n) {
                res.add("pushback a[${r - 1}]")
                res.add("min")
                r++
            }
            d = -1
        } else {
            var r = n
            while (r > l) {
                res.add("min")
                res.add("popback")
                r--
            }
            res.add("min")
            d = 1
        }

        if (l < n) {
            res.add("popfront")
        }
    }
    return res
}

fun main() {
    val fs = FastScanner()
    val n = fs.nextInt()
    val ans = solve(n)
    val out = StringBuilder()
    out.append(ans.size).append('\n')
    for (cmd in ans) {
        out.append(cmd).append('\n')
    }
    print(out.toString())
}

